iOS/TvShows/TvShows/Modules/Login/LoginPresenter.swift

```diff
@@ -28,15 +29,91 @@ final class LoginPresenter {
 extension LoginPresenter: LoginPresenterInterface {
 
     func configure(with output: Login.ViewOutput) -> Login.ViewInput {
-        let areButtonsAvailable = onButtonsAvailable(email: output.email, password: output.password)
+        handle(
+            login: output.login,
+            output.email,
+            output.password
+        )
+        handle(
+            register: output.register,
+            output.email,
+            output.password
+        )
 
+        let areButtonsAvailable = onButtonsAvailable(email: output.email, password: output.password)
         return Login.ViewInput(areButtonsAvailable: areButtonsAvailable)
     }
 }
 
 extension LoginPresenter {
-    // TODO: Implement onButtonsAvailable
+
+    func handle(
+        login: Signal<Void>,
+        _ email: Driver<String?>,
+        _ password: Driver<String?>
+    ) {
+        let inputs = Driver.combineLatest(email.compactMap { $0 }, password.compactMap { $0 })
+        login
+            .withLatestFrom(inputs)
+            .flatMap { [unowned self] email, password -> Driver<Member> in
+                performLogin(email, password)
+            }
+            .do(onNext: { member in
+                print("Successfully logged in member: \(member)")
+            })
+            .drive(onNext: { [unowned wireframe] _ in
+                wireframe.navigateToHome()
+            })
+            .disposed(by: disposeBag)
+    }
+
+    func performLogin(_ email: String, _ password: String) -> Driver<Member> {
+        interactor
+            .login(with: email, password)
+            .do(onError: { _ in
+                // TODO: Hide progress indicator and show validation error
+            }, onSubscribe: {
+                // TODO: Show progress indicator
+            })
+            .asDriver(onErrorDriveWith: .never())
```

Comment:
It would be better to return `.empty()` here rather than never. 
It's better for the flatMap chain to complete than stay alive on error.